submissions/servers/servers.md

```diff
@@ -0,0 +1,47 @@
+
+This whole idea of Lockstep and Fully Deterministic algorithms lessens the probability of a [desynchronization](https://wiki.factorio.com/Desynchronization) (desync) happening, but the chance is still there. Normally, when the client and server are doing the same thing at the same time, they are happy, as they are synchronized (in sync). When they fall out of sync, a desync happens. A desync can occur due to two clients calculating an update with different results. During a desync, a player is instantly disconnected from the game and starts downloading the map the server has to include in a desync report, which can then be useful to mod creators or Factorio devs, depending on what causes the desync. See the image below for an example.
+
+{% include image.html src='desync-chat.png' %}
+
+You may wonder, how do desyncs not occur with robots moving across the map? Surely if they all do tasks and some robots are chosen to do said tasks, different clients might choose different robots to perform those tasks, no? Nope. Somewhere in the game's C++ code, there is a list of inactive robots (robots returning to roboports). If there aren't any close to the site of the building, the game dispatches a 'fresh' robot from a roboport. The ghosts are iterated over sequentially - they are put in a list, probably according to the XY coordinate of the entity, and the robot with the shortest path to that entity is dispatched (from roboport to logistic chest to entity). Keep in mind, this information **is an incomplete thought** of *curiosity* as nobody (except for the game developers) knows how this works exactly.
+
+This has been only one example of many things that can go wrong with multiplayer: robot construction, biter AI simulations, and even mod things.
+
+Even if you would want to use something as simple as `math.random()` in a mod or scenario of yours, there would be consistent results - all clients would get the same result of the function. If you know a bit of programming, you may understand seeding a random function. If not, seeding a random function is done with a so-called 'random seed', which initializes the random number generator. It is important to note it is a *pseudorandom* generator, and therefore not truly random, as it is initialized with a pre-determined number, which allows the results to be reproduced anywhere. See [this](https://en.wikipedia.org/wiki/Random_seed). According to [this StackOverflow answer](https://softwareengineering.stackexchange.com/a/298616), a pseudorandom number generator works by seeding it with a number first. Then, you would multiply the said number with an [extremely large prime number](https://primes.utm.edu/primes/search.php?Number=100). The number that is used as a seed is then increased by 1, which results in a completely different number after multiplication.
+
+I couldn't explain everything about this in one simple post. If you would want to learn more about this, you should probably visit one of these:
```

Comment:
I think if these links are somewhere in your article already then it would be better to not add it here too :)